Adopt some more smart pointers in UIProcess

rwlbuis opened the pull request:
#### 1ab7168d3ed16b9eb76a0407530dfea65cff2da8
<pre>
Adopt some more smart pointers in UIProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=297652">https://bugs.webkit.org/show_bug.cgi?id=297652</a>

Reviewed by Ryosuke Niwa.

Fix more [alpha.webkit.UnretainedCallArgsChecker] warnings in UIProcess.

* Source/WebKit/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/mac/TiledCoreAnimationDrawingAreaProxy.mm:
(WebKit::TiledCoreAnimationDrawingAreaProxy::createFence):
* Source/WebKit/UIProcess/mac/WKPrintingView.h:
* Source/WebKit/UIProcess/mac/WKPrintingView.mm:
(-[WKPrintingView _delayedResumeAutodisplay]):
* Source/WebKit/UIProcess/mac/WKRevealItemPresenter.mm:
(-[WKRevealItemPresenter initWithWebViewImpl:item:frame:menuLocation:]):
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(WebKit::WebContextMenuProxyMac::setupServicesMenu):
* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
* Source/WebKit/UIProcess/mac/WebDateTimePickerMac.mm:
(-[WKDateTimePicker initWithParams:inView:]):
(-[WKDateTimePicker showPicker:]):
(-[WKDateTimePicker invalidate]):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKResponderChainSink initWithResponderChain:]):

Canonical link: <a href="https://commits.webkit.org/300923@main">https://commits.webkit.org/300923@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a9a19eab453f733dcc14a03f236bb6bb8b8bbd80

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124240 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43927 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34650 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131074 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76320 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44670 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52529 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94505 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62697 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127194 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35595 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111126 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75093 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34541 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74554 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105338 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29509 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133745 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51156 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38998 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102981 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51545 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107344 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102781 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26173 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48145 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26392 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48064 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51018 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56797 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50462 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53816 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52137 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/mac/WKFullScreenWindowController.mm:
```diff
@@ -299,19 +299,19 @@ - (BOOL)isFullScreen
 }
 
-- (WebCoreFullScreenPlaceholderView *)webViewPlaceholder
+- (RetainPtr<WebCoreFullScreenPlaceholderView>)webViewPlaceholder
```
It is fishy to be changing getters to return a RetainPtr, we wouldn't do this in C++ for example due to ref-count churn. Why can't the call sites construct a RetainPtr when they need one?

rwlbuis replied:
I think fixing the call sites would not get rid of the warning:
https://ews-build.s3-us-west-2.amazonaws.com/macOS-Safer-CPP-Checks-EWS/fdf57ecf-51467/scan-build-output/StaticAnalyzerRegressions/WebKitLegacy/index.html

Now, that may be a bogus warning, since my new PR shows the same kind of warning is not given for a method that does the same thing with same return type.
I'll have a look whether clang-tidy can be fixed to not issue this warning.

cdumez commented on Source/WebKit/UIProcess/mac/WKFullScreenWindowController.mm:
```diff
@@ -299,19 +299,19 @@ - (BOOL)isFullScreen
 }
 
-- (void)setSavedConstraints:(NSArray *)savedConstraints
+- (void)setSavedConstraints:(RetainPtr<NSArray>)savedConstraints
```
Same

cdumez commented on Source/WebKit/UIProcess/mac/WKFullScreenWindowController.mm:
```diff
@@ -299,19 +299,19 @@ - (BOOL)isFullScreen
 }
 
-- (NSArray *)savedConstraints
+- (RetainPtr<NSArray>)savedConstraints
```
Same

rniwa commented on Source/WebKit/UIProcess/mac/WKFullScreenWindowController.h:
```diff
@@ -71,11 +71,11 @@ typedef enum FullScreenState : NSInteger FullScreenState;
 
 @property (readonly) NSRect initialFrame;
 @property (readonly) NSRect finalFrame;
-@property (assign) NSArray *savedConstraints;
```
I think we wanna keep the property as a raw pointer value and change its corresponding ivar to be `RetainPtr`.

rwlbuis replied:
I removed that part.

rniwa commented on Source/WebKit/UIProcess/mac/WKFullScreenWindowController.h:
```diff
@@ -71,11 +71,11 @@ typedef enum FullScreenState : NSInteger FullScreenState;
 
 - (instancetype)initWithWindow:(NSWindow *)window webView:(WKWebView *)webView page:(std::reference_wrapper<WebKit::WebPageProxy>)page;
 
-@property (nonatomic, readonly) WebCoreFullScreenPlaceholderView *webViewPlaceholder;
```
Ditto.

rwlbuis replied:
Also removed.

rwlbuis committed:
Adopt some more smart pointers in UIProcess
https://bugs.webkit.org/show_bug.cgi?id=297652

Reviewed by Ryosuke Niwa.

Fix more [alpha.webkit.UnretainedCallArgsChecker] warnings in UIProcess.

* Source/WebKit/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/mac/TiledCoreAnimationDrawingAreaProxy.mm:
(WebKit::TiledCoreAnimationDrawingAreaProxy::createFence):
* Source/WebKit/UIProcess/mac/WKPrintingView.h:
* Source/WebKit/UIProcess/mac/WKPrintingView.mm:
(-[WKPrintingView _delayedResumeAutodisplay]):
* Source/WebKit/UIProcess/mac/WKRevealItemPresenter.mm:
(-[WKRevealItemPresenter initWithWebViewImpl:item:frame:menuLocation:]):
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(WebKit::WebContextMenuProxyMac::setupServicesMenu):
* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
* Source/WebKit/UIProcess/mac/WebDateTimePickerMac.mm:
(-[WKDateTimePicker initWithParams:inView:]):
(-[WKDateTimePicker showPicker:]):
(-[WKDateTimePicker invalidate]):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKResponderChainSink initWithResponderChain:]):

Canonical link: https://commits.webkit.org/300923@main
